events/voiceStateUpdate.js

```diff
@@ -158,6 +158,12 @@ module.exports = {
 			await player.musicHandler.disconnect();
 			return;
 		}
+		// do not set pause timeout when a stage ends
+		if (oldState.channel.type === 'GUILD_STAGE_VOICE') {
+			if (await !player) return;
+			if (await !player?.connected) return;
+			if (await !oldState.channel.members.get(bot.user.id)) return;
+		}
```

Comment:
i'm still positive there's a way to do this without abusing await, just not sure how yet